File: frontend/src/types/resumenZona.ts
export interface ResumenZonaFila {
  zona: string;
  tipo_brigada: string;
  brigadas_reportadas: number;
  brigadas_contrato: number;
  porcentaje_brigadas_efectivas: number;
  reconexiones_programadas: number;
  total_reconexiones_ejecutadas: number;
  promedio_reconexiones: number;
  asignacion_carga: number;
  corte_programado: number;
  total_cortes: number;
  cumplimiento_corte_porcentaje: number;
  promedio_cortes: number;
  total_actividades: number;
  promedio_actividades: number;
  cumplimiento_promedio_meta: number;
  observacion: string;
}

export interface ResumenZonaResponse {
  fecha_operacional: string;
  zonas: ResumenZonaFila[];
  totales: ResumenZonaFila[];
  alertas: string[];
}

File: backend/app/schemas/resumen_zona.py
from pydantic import BaseModel
from typing import List, Optional


class ResumenZonaFila(BaseModel):
    zona: str
    tipo_brigada: str
    brigadas_reportadas: int
    brigadas_contrato: int
    porcentaje_brigadas_efectivas: float
    reconexiones_programadas: int
    total_reconexiones_ejecutadas: int
    promedio_reconexiones: float
    asignacion_carga: int
    corte_programado: int
    total_cortes: int
    cumplimiento_corte_porcentaje: float
    promedio_cortes: float
    total_actividades: int
    promedio_actividades: float
    cumplimiento_promedio_meta: float
    observacion: str


class ResumenZonaResponse(BaseModel):
    fecha_operacional: str
    zonas: List[ResumenZonaFila]
    totales: List[ResumenZonaFila]
    alertas: List[str]
